cpp/src_prims/sparse/coo.h

```diff
@@ -296,57 +263,53 @@ cusparseStatus_t cusparse_gthr(cusparseHandle_t handle, int nnz, double *vals,
  * @param rows rows array from coo matrix
  * @param cols cols array from coo matrix
  * @param vals vals array from coo matrix
+ * @param alloc device allocator for temporary buffers
  * @param stream: cuda stream to use
  */
 template <typename T>
 void coo_sort(int m, int n, int nnz, int *rows, int *cols, T *vals,
-              cudaStream_t stream = 0) {
+              std::shared_ptr<deviceAllocator> alloc, cudaStream_t stream) {
   cusparseHandle_t handle = NULL;
 
   size_t pBufferSizeInBytes = 0;
-  void *pBuffer = NULL;
-  int *d_P = NULL;
 
   CUSPARSE_CHECK(cusparseCreate(&handle));
-
   CUSPARSE_CHECK(cusparseSetStream(handle, stream));
-
   CUSPARSE_CHECK(cusparseXcoosort_bufferSizeExt(handle, m, n, nnz, rows, cols,
                                                 &pBufferSizeInBytes));
 
-  allocate(d_P, nnz);
-  cudaMalloc(&pBuffer, pBufferSizeInBytes * sizeof(char));
+  device_buffer<int> d_P(alloc, stream, nnz);
+  device_buffer<char> pBuffer(alloc, stream, pBufferSizeInBytes);
 
-  CUSPARSE_CHECK(cusparseCreateIdentityPermutation(handle, nnz, d_P));
+  CUSPARSE_CHECK(cusparseCreateIdentityPermutation(handle, nnz, d_P.data()));
 
-  CUSPARSE_CHECK(
-    cusparseXcoosortByRow(handle, m, n, nnz, rows, cols, d_P, pBuffer));
+  CUSPARSE_CHECK(cusparseXcoosortByRow(handle, m, n, nnz, rows, cols,
+                                       d_P.data(), pBuffer.data()));
 
-  T *vals_sorted;
-  allocate(vals_sorted, nnz);
+  device_buffer<T> vals_sorted(alloc, stream, nnz);
 
-  CUSPARSE_CHECK(cusparse_gthr<T>(handle, nnz, vals, vals_sorted, d_P));
+  CUSPARSE_CHECK(
+    cusparse_gthr<T>(handle, nnz, vals, vals_sorted.data(), d_P.data()));
 
   cudaDeviceSynchronize();
```

Comment:
```suggestion
  CUDA_CHECK(cudaStreamSynchronize(stream));
```

Reply:
ph wow. This one was left around from a long time ago and looked right over it in the refactor. Good find!